### DISCUSSION TITLE
Auto increment on Number field - PostgreSQL

### DISCUSSION THREAD

**Discussion by gonzarascon** [strapi/strapi#6631] [Legacy Topics]
Hi! 

Is there any way to handle auto incremental number attribute on Strapi Content-Type? (I'm using PostgreSQL as db)

**Comment by basavarajdodamani** [comment]
Hi @gonzarascon , Even i am in need of implementing something similar with MongoDB as backend. I do not think it is possible at the moment, as per below article: https://github.com/strapi/strapi/issues/861
Were you able to find any alternative?

Most recent thread: https://github.com/strapi/strapi/issues/2784

**Reply by gonzarascon** [reply]
I just used the beforeCreate [lifecycle hook](https://strapi.io/documentation/v3.x/concepts/models.html#lifecycle-hooks) on that model. It feels dirty, but does the trick.

**Reply by basavarajdodamani** [reply]
> 
> 
> I just used the beforeCreate [lifecycle hook](https://strapi.io/documentation/v3.x/concepts/models.html#lifecycle-hooks) on that model. It feels dirty, but does the trick.

Okay. Yes i have used this feature before but for other purpose, Could you please let me know what did you write inside this beforeCreate? You would need to have access to last record to increment.

**Reply by gonzarascon** [reply]
Sure: 
```javascript 

   lifecycles: {
    async beforeCreate(data) {
     // We fetch all entries
      const allEntries = await strapi.query("entry").find();

     // Using lodash here just to simplify things:
    // We want to descentantlly order Entries array by 
    // the property we want to auto-increment so we
   // can access to that value at the first position of the array

      const lastEntryNumber = _.orderBy(
        allEntries ,
        ["entry_number"],
        ["desc"]
      );

  // We access that last value and add it 1 (if you are auto-incrementing by 1) and set the result as the new entry property.
      data.entry_number = parseInt(lastEntryNumber [0].entry_number) + 1;
    }
 }

```

**Reply by derrickmehaffy** [reply]
you could pass along some filters to get the latest also `strapi.query('entry').find({ limit: 1, sort: 'entry_number:desc'})` or something also
